### ISSUE TITLE
Unity false positive RCS1213, RCS1169

### ISSUE THREAD

**Issue by ADIX7** [JosefPihrt/Roslynator#585]
I'm working on a Unity project, I managed to add analyzers to it. Unfortunately Unity has some wierd thing.
* A developer can use 'lifecycle' methods which are private void (by convention) methods you have to implement in your classes. They are called by C++ methods. Unfortunately RCS1213 would remove these methods because they are not called anywhere.
* Another thing is private fields which are also set from outside (Unity magic). RCS1169 tells me they should be readonly.

I use the newest Roslynator nuget packages.

I can make a PR if you wish.

**Comment by miltoncandelero** [comment]
I am having the same issues and also having issues with IDE0005 (that is the same as RCS1213). Maybe we should make rulesets for specifics workflows? I am using the VSCode extensions btw.

**Comment by JosefPihrt** [comment]
Common pattern is that you have a base ruleset that is used for a solution and then you have "derived" rulesets that inherits rules from base ruleset.

So I would recommend to change your example in the following manner:

```xml

<?xml version="1.0" encoding="utf-8"?>
<RuleSet Name="Unity Rules" ToolsVersion="15.0">

  <Include Path="PATH_TO_BASE_RULESET" Action="Default" />

  <Rules AnalyzerId="Roslynator.CSharp.Analyzers" RuleNamespace="Roslynator.CSharp.Analyzers">
    <Rule Id="RCS1213" Action="None" />
    <Rule Id="RCS1169" Action="None" />
  </Rules>

  <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp.Features" RuleNamespace="Microsoft.CodeAnalysis.CSharp.Features">
    <Rule Id="IDE0051" Action="None" />
  </Rules>

</RuleSet>
```

**Comment by SamMacedo** [comment]
Another way could be by using an `.editorconfig` file, see my comment here: https://github.com/OmniSharp/omnisharp-vscode/issues/3174#issuecomment-666901713

**Comment by Molder72** [comment]
**For unity users**
as it was not easy to find the information, I leave a memo here.
overwrite the file .roslynatorconfig
<html>
<body>


OS | Path
-- | --
Windows | C:/Users/<USERNAME>/AppData/Local/.roslynatorconfig
Linux | /home/<USERNAME>/.local/share/.roslynatorconfig
OSX | /Users/<USERNAME>/.local/share/.roslynatorconfig


</body>
</html>

```
# Roslynator Config File

is_global = true

# Options in this file can be used to change default configuration of analyzers, refactorings and compiler diagnostic fixes.
# Default configuration is loaded once when IDE starts. Therefore, it may be necessary to restart IDE for changes to take effect.
# Full list of available options: https://github.com/JosefPihrt/Roslynator/blob/master/docs/Configuration.md
roslynator_suppress_unity_script_methods = true # resolve Start() OnEnable() and all other unity methos that by defaul it's show transparent.

# Set severity for all analyzers that are enabled by default (https://docs.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2022#set-rule-severity-of-multiple-analyzer-rules-at-once-in-an-editorconfig-file)
dotnet_analyzer_diagnostic.category-roslynator.severity = default|none|silent|suggestion|warning|error

# Enable/disable all analyzers by default.
# NOTE: This option can be used only in .roslynatorconfig file
roslynator_analyzers.enabled_by_default = true|false

## Set severity for a specific analyzer
#dotnet_diagnostic.<ANALYZER_ID>.severity = default|none|silent|suggestion|warning|error
dotnet_diagnostic.RCS1213.severity = none # supported from standart visual stucio 2022 IDE0051 by default
dotnet_diagnostic.RCS1169.severity = none

## Enable/disable all refactorings
#roslynator_refactorings.enabled = true|false

## Enable/disable specific refactoring
#roslynator_refactoring.<REFACTORING_NAME>.enabled = true|false

## Enable/disable all compiler diagnostic fixes
#roslynator_compiler_diagnostic_fixes.enabled = true|false

## Enable/disable specific compiler diagnostic fix
#roslynator_compiler_diagnostic_fix.<COMPILER_DIAGNOSTIC_ID>.enabled = true|false
```
re-start Visual Studio to apply the changes.

Unity: Version 2021.3.0
Visual Studio 2022: Version 17.1.4
Roslynator 2022 extension installed in visual studio:  Version 4.1.0

**Comment by JosefPihrt** [comment]
Hi @Molder72,

Actually it is not necessary to disable RCS1213 and RCS1169. It is sufficient to set following option:

```editorconfig
roslynator_suppress_unity_script_methods = true
```
